### DISCUSSION TITLE
Plans for real time montage viewing?

### DISCUSSION THREAD

**Discussion by forkineye** [blakeblackshear/frigate#1339] [General]
Are there any plans for real-time viewing of multiple cameras, ala montage?  If not, what setups are you all using for this?  Would like to migrate to Frigate as my total NVR solution and ditch Zoneminder all together if possible.

**Comment by blakeblackshear** [comment]
Next release includes a Birdseye view feature. It is a live, dynamic grid view of all cameras with activity. It has 3 modes:

- objects: includes any cameras with detected objects
- motion: includes any cameras with detected motion
- continuous: all cameras all the time

It is designed to minimize the bandwidth used, so when there are no active cameras, you just get a black screen with the frigate logo.

**Reply by therealcarnivor** [reply]
with 9 cameras that i have, this is fantastic in continuous for me, any plans to be able to have a birdseye camera stream available in HA too??

**Reply by trythatagain** [reply]
Really nice, I like the continuous view, thats the only reason I'm still using Shinobi as I don't run HA. 
Will there be any new interfaces for viewing clips and continuous recordings?

Is the next version stable enough to start using and trying to find bugs etc?

**Reply by Hankanman** [reply]
Quick suggestion on this feature, can we have a role for cameras that are included? I have 5 cameras but one of them i only use for detection and would rather have the 4 "proper" cameras only in the birdseye view

**Reply by blakeblackshear** [reply]
@hankanman please open an actual feature request

**Reply by DT00689** [reply]
Is this available in Home Assistant add on?

**Reply by blakeblackshear** [reply]
No, but the frigate lovelace card should support it in the future.

**Comment by mattheys** [comment]
I did this https://github.com/blakeblackshear/frigate/discussions/490 to get 4 cameras on a single web page, I also use https://github.com/aler9/rtsp-simple-server to proxy my cameras before going to Frigate to save bandwidth with multiple services accessing the streams, it now has an HLS option. 

I use tinycam free on Android to just subscribe to each stream and get them all at once on my phone.

I also did this beast of a query on my nvidia jetson nano to get a single stream mosaic but I had issues because if a camera dropped it's connection briefly this didn't reatach the stream it just went blank.
```
VUSR=user
VPASS=pass
WIDTH=1920
HEIGHT=1080

gst-launch-1.0 \
  nvcompositor name=comp \
  sink_0::xpos=0 sink_0::ypos=0 sink_0::width=$WIDTH sink_0::height=$HEIGHT \
  sink_1::xpos=$WIDTH sink_1::ypos=0 sink_1::width=$WIDTH sink_1::height=$HEIGHT \
  sink_2::xpos=0 sink_2::ypos=$HEIGHT sink_2::width=$WIDTH sink_2::height=$HEIGHT \
  sink_3::xpos=$WIDTH sink_3::ypos=$HEIGHT sink_3::width=$WIDTH sink_3::height=$HEIGHT ! \
  nvvidconv ! 'video/x-raw(memory:NVMM), format=(string)I420' ! \
  omxh265enc bitrate=500000 control-rate=constant vbv-size=1 ! \
  rtspclientsink name=sink location=rtsp://192.168.1.90:8554/myStream  \
  rtspsrc location=rtsp://$VUSR:$VPASS@192.168.1.90:8554/garage latency=0 drop_on_latency=true ! rtph264depay ! queue ! h264parse ! nvv4l2decoder ! comp. \
  rtspsrc location=rtsp://$VUSR:$VPASS@192.168.1.90:8554/porch latency=0 drop_on_latency=true ! rtph264depay ! queue ! h264parse ! nvv4l2decoder ! comp. \
  rtspsrc location=rtsp://$VUSR:$VPASS@192.168.1.90:8554/back latency=0 drop_on_latency=true ! rtph264depay ! queue ! h264parse ! nvv4l2decoder ! comp. \
  rtspsrc location=rtsp://$VUSR:$VPASS@192.168.1.90:8554/nursery latency=0 drop_on_latency=true ! rtph264depay ! queue ! h264parse ! nvv4l2decoder ! comp.
```
